holmes/plugins/toolsets/bash/bash_toolset.py

```diff
@@ -0,0 +1,303 @@
+                error=f"The 'command' parameter must be a string, got {type(command_str).__name__}.",
+                params=params,
+            )
+        try:
+            safe_command_str = make_command_safe(command_str, self.toolset.config)
+        except (argparse.ArgumentError, ValueError) as e:
+            logging.info(f"Refusing LLM tool call {command_str}", exc_info=True)
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                error=f"Refusing to execute bash command. Only some commands are supported and this is likely because requested command is unsupported. Error: {str(e)}",
+                params=params,
+            )
+
+        try:
+            process = subprocess.run(
+                safe_command_str,
+                shell=True,
+                executable="/bin/bash",
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT,
+                text=True,
+                timeout=timeout,
+                check=False,
+            )
+
+            result_data = (
+                f"{command_str}\n" f"{process.stdout.strip() if process.stdout else ''}"
+            )
+
+            status = (
+                ToolResultStatus.SUCCESS
+                if process.returncode == 0
+                else ToolResultStatus.ERROR
+            )
+
+            return StructuredToolResult(
+                status=status,
+                data=result_data,
+                params=params,
+                invocation=safe_command_str,
+                return_code=process.returncode,
+            )
+        except subprocess.TimeoutExpired:
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                data=f"Error: Command '{command_str}' timed out after {timeout} seconds.",
+                params=params,
+            )
+        except FileNotFoundError:
+            # This might occur if /bin/bash is not found, or if shell=False and command is not found
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                data="Error: Bash executable or command not found. Ensure bash is installed and the command is valid.",
+                params=params,
+            )
+        except Exception as e:
+            return StructuredToolResult(
+                status=ToolResultStatus.ERROR,
+                data=f"Error executing command '{command_str}': {str(e)}",
+                params=params,
+            )
+
+    def get_parameterized_one_liner(self, params: Dict[str, Any]) -> str:
+        command = params.get("command", "N/A")
+        display_command = command[:200] + "..." if len(command) > 200 else command
+        return display_command
+
+
+class BashExecutorToolset(BaseBashExecutorToolset):
+    def __init__(self):
+        super().__init__(
+            name="bash",
+            enabled=True,  # Default state; can be overridden by global config.
```

Comment:
I remebmer that we said that this will be disabled by default. Am i wrong? if yes ignore.

Reply:
Indeed.